apps/portal/src/app/graphql/graphql.service.ts

```diff
@@ -51,32 +85,32 @@ export class GraphqlService {
 
   // eslint-disable-next-line @typescript-eslint/no-explicit-any
   mutate<T>(mutation: DocumentNode, variables: any): Observable<MutationResult<T>> {
-    const token = this.getToken();
-    return this.apollo
-      .mutate<T>({
-        mutation,
-        variables: { file: variables.inputFile },
-        context: {
-          hasUpload: true,
-          useMultipart: true,
-          headers: {
-            'content-type': 'application/json',
-            'x-apollo-operation-name': 'createSummary',
-            Authorization: `Bearer ${token}`,
+    return this.getToken().pipe(
+      switchMap((token: string) =>
+        this.apollo.mutate<T>({
+          mutation,
+          variables: { file: variables.inputFile },
+          context: {
+            hasUpload: true,
+            useMultipart: true,
+            headers: {
+              'content-type': 'application/json',
+              'x-apollo-operation-name': 'createSummary',
+              Authorization: `Bearer ${token}`,
+            },
           },
-        },
-        errorPolicy: 'all',
-        fetchPolicy: 'network-only',
-      })
-      .pipe(
-        catchError((error) => {
-          if (this.isUnauthorizedError(error)) {
-            this.handleUnauthorizedError();
-          }
-
-          return throwError(error);
+          errorPolicy: 'all',
+          fetchPolicy: 'network-only',
         }),
-      );
+      ),
+      catchError((error) => {
+        if (this.isUnauthorizedError(error)) {
+          this.handleUnauthorizedError();
+        }
+
+        return throwError(error);
+      }),
+    );
   }
```

Comment:
Make this change
`apps/api/src/Decorator`
Move this to common folder and folder name should be `decorators` casing same as others and plural.

Reply:
I didn't understand can you please clarify this these changes are done in the portal

Reply:
Updated as per my understand in this comment : https://github.com/OsmosysSoftware/osm-transcript-summarizer/pull/40/commits/c2f87a6106d0cbdf1ec18c0ef5c8d1303edacbef